pyproject.toml

```diff
@@ -70,19 +70,23 @@ array-api-extra = { path = ".", editable = true }
 
 [tool.pixi.feature.lint.dependencies]
 pre-commit = "*"
-mypy = "*"
 pylint = "*"
 # import dependencies for mypy:
 array-api-strict = "*"
 numpy = "*"
 pytest = "*"
 
+[tool.pixi.feature.lint.pypi-dependencies]
+basedmypy = "*"
+basedpyright = "*"
+
 [tool.pixi.feature.lint.tasks]
 pre-commit-install = { cmd = "pre-commit install" }
 pre-commit = { cmd = "pre-commit run -v --all-files --show-diff-on-failure" }
 mypy = { cmd = "mypy", cwd = "." }
 pylint = { cmd = ["pylint", "array_api_extra"], cwd = "src" }
-lint = { depends-on = ["pre-commit", "pylint", "mypy"] }
+pyright = { cmd = "basedpyright", cwd = "." }
+lint = { depends-on = ["pre-commit", "pylint", "mypy", "pyright"] }
```

Comment:
you could also consider adding mypy and pyright to the pre-commit config:

https://github.com/pre-commit/mirrors-mypy
https://docs.basedpyright.com/latest/installation/pre-commit%20hook/